test/unit/integrations/coinbase/coinbase_return_test.rb

```diff
@@ -23,12 +23,24 @@ def test_invalid_return
     assert coinbase_return.success?
   end
 
+  def test_catch_nil_params
+    Net::HTTP.any_instance.expects(:request).returns(stub(:body => invalid_http_raw_data))
+    coinbase_return = Coinbase::Return.new(valid_query_string, @options)
+
+    assert !coinbase_return.notification.acknowledge
+    assert coinbase_return.success?
+  end
+
   private
 
   def valid_http_raw_data
     '{"order":{"id":"OQJ836AF","created_at":"2014-07-09T07:00:33-07:00","status":"completed","event":{"type":"completed"},"total_btc":{"cents":1463,"currency_iso":"BTC"},"total_native":{"cents":1,"currency_iso":"CAD"},"total_payout":{"cents":0,"currency_iso":"USD"},"custom":"10","receive_address":"19FuVxoEvVLxRibVnmSciNXEsfgFs8W29Z","button":{"type":"buy_now","name":"Shop One - #10","description":null,"id":"0c3e9c6dd38619a2ba11b4561631e6ad"},"refund_address":"1BmmrdqcLqGCtx54vvencNS8VCMsjJCEBA","transaction":{"id":"53bd4b0e86fd2456d8000003","hash":null,"confirmations":0}}}'
   end
 
+  def invalid_http_raw_data
+    'params { order: { id: "81ILZDAZ", uuid: "72880bce-c70a-5e89-bba9-ebd2ba0e554c", resource_path: "/v2/orders/72880bce-c70a-5e89-bba9-ebd2ba0e554c", metadata: null, created_at: "2015-10-28T11:00:24-07:00", status: "completed", event: { type: "completed" }, total_btc: { cents: 4617000, currency_iso: "BTC" }, total_native: { cents: 900, currency_iso: "GBP" }, total_payout: { cents: 1378, currency_iso: "USD" }, custom: "4467632773", receive_address: "1CHwsxMfLhJ91hZ3iv8BuHdgJJdZEtRn5F", button: { type: "buy_now", subscription: null, repeat: null, name: "M S Canada - #4467632773", description: null, id: "cc64986cd3af4139ed0c25c46bece43d", uuid: "10ab7364-f9e9-59b7-9ce1-6eb9607ad9a3", resource_path: "/v2/checkouts/10ab7364-f9e9-59b7-9ce1-6eb9607ad9a3" }, discount_percent: 10, total_original: { cents: 1000, currency_iso: "GBP" }, transaction: { id: "56310daf18179c0001012d9a", hash: "2cad8dfa2a9306b2e8a5b721d471502c6942d003dd721237b11a6f298f7cc0c8", confirmations: 0 } }, controller: "services/ping", action: "notify_integration", gateway: "coinbase", shop_id: "2" }'
```

Comment:
Although the test passes, it's not going through the code path that you fixed. This `invalid_http_raw_data` you added is actually invalid JSON, so ends up going through [this rescue block](https://github.com/activemerchant/offsite_payments/blob/2982c6dc5987b75edc8d2e26f9cd4d3ce5f9fa87/lib/offsite_payments/integrations/coinbase.rb#L124-L126). Instead what we want is valid json that's missing an order key.

The easiest way to do this is to  parse `valid_http_raw_data`, reject the `order` key and call `to_json` on it:
```
def http_raw_data_missing_order_key
  JSON.parse(valid_http_raw_data).reject { |key, value| key == 'order' }.to_json
end
```